1-js/05-data-types/03-string/article.md

```diff
@@ -266,15 +266,15 @@ while (true) {
 }
 ```
 
-The same algorithm can be layed out shorter:
+The same algorithm can be layed out shorter and more efficient:
 
 ```js run
 let str = "As sly as a fox, as strong as an ox";
 let target = "as";
 
 *!*
 let pos = -1;
-while ((pos = str.indexOf(target, pos + 1)) != -1) {
+while ((pos = str.indexOf(target, pos += target.length)) != -1) {
```

Comment:
Maybe:

```suggestion
while ((pos = str.indexOf(target, pos + target.length)) != -1) {
```

It seems there is no reason to reassign `pos` twice here.

Reply:
You're absolutely right @vsemozhetbyt. I'm so used to `+=`, it was basically a reflex. Thanks!